### PR TITLE
Core Service as a Sink ref case added

### PR DESCRIPTION
**Pull request by tzununbekov**
Kubernetes core Service is a special case of Sinks. This PR adds an explanation of how destination ports are being chosen from single- and multi-port Service destinations.

## Proposed Changes

- `v1.Service` sink description added to the eventing/sinks doc

Do not merge until knative/pkg#2322 is merged.

### CODE REVIEW

File: docs/eventing/sinks/README.md [hunk 1]
```diff
@@ -55,6 +55,21 @@ spec:
...
+If Sink `ref` points to the Kubernetes core Service object, the destination port is chosen according to the following logic:
+
+- `knative.dev/destination-port` Service annotation can explicitly pin the port name that should be used as a destination,
+- Otherwise, if port annotation is not set,
+
+    \- _single-port_ Service's only port will be used as a destination,
+
+    \- _multi-port_ Service's port 80 will be used if it is exposed by the service.
+
+In all other cases or if the port specified in `knative.dev/destination-port` annotation does not exist in the service, sink URI resolution will fail.
```
Comment (snneji) [inline docs/eventing/sinks/README.md:69]:
```suggestion
In all other cases or if the port specified in `knative.dev/destination-port` annotation does not exist in the service, sink URI resolution will fail.
```
```suggestion
In all other cases, or if the port specified in `knative.dev/destination-port` annotation does not exist in the Service, Sink URI resolution fails.
```

File: docs/eventing/sinks/README.md [hunk 2]
```diff
@@ -55,6 +55,21 @@ spec:
...
+
+The default protocol used in Sink URL is HTTP unless the destination port is 443 - in that case, HTTPS will be used as a URL scheme.
```
Comment (snneji) [inline docs/eventing/sinks/README.md:71]:
```suggestion
The default protocol used in Sink URL is HTTP unless the destination port is 443 - in that case, HTTPS will be used as a URL scheme.
```
```suggestion
The default protocol used in Sink URL is HTTP. If the destination port is 443, HTTPS is used as a URL scheme.
```

File: docs/eventing/sinks/README.md [hunk 3]
```diff
@@ -55,6 +55,21 @@ spec:
...
+## Kubernetes Service as a sink
+
+If Sink `ref` points to the Kubernetes core Service object, the destination port is chosen according to the following logic:
+
+- `knative.dev/destination-port` Service annotation can explicitly pin the port name that should be used as a destination,
+- Otherwise, if port annotation is not set,
+
+    \- _single-port_ Service's only port will be used as a destination,
+
+    \- _multi-port_ Service's port 80 will be used if it is exposed by the service.
```
Comment (snneji) [inline docs/eventing/sinks/README.md:67]:
```suggestion
    \- _single-port_ Service's only port will be used as a destination,

    \- _multi-port_ Service's port 80 will be used if it is exposed by the service.
```
```suggestion
    - For _single-port_: the Service's only port is used as a destination.

    - For _multi-port_: the Service's port 80 is used if it is exposed by the service.
```

**Review by snneji — CHANGES_REQUESTED** [review]

**Comment by snneji** [comment]
Just a few style tweaks, otherwise looks good to me.

**Comment by tzununbekov** [comment]
@snneji thanks for the review, I've applied your suggestions. I believe this change must be on hold until knative/pkg#2322  is merged.

/hold

**Review by snneji — APPROVED** [review]

**Comment by abrennan89** [comment]
@tzununbekov any update on this one? It looks like the related pull request hasn't been updated in some time.

**Comment by tzununbekov** [comment]
@abrennan89 yeah, I didn't have much time to work on this lately. Planning to get back to these PRs later this week.

**Comment by dprotaso** [comment]
I resolved the conflict via the GitHub UI since it was trivial enough.

/lgtm

**Comment by dprotaso** [comment]
Looks like the dependent pkg PR is still open - so keeping the hold

**Comment by csantanapr** [comment]
after 3 months it looks this work is abandoned in https://github.com/knative/pkg/pull/2322
Feel free to reopen or open a new PR once the feature is implemented @tzununbekov

**Pull request closed without merging**